pxd.c

```diff
@@ -2097,6 +2101,7 @@ static int pxd_control_open(struct inode *inode, struct file *file)
 	file->private_data = fc;
 
 	pxdctx_set_connected(ctx, true);
+	wmb();
```

Comment:
memory barrier is redundant here since spin_unlock just happened which does memory barrier

Reply:
allow_disconnected is outside lock. In any case, even if wmb() is redundant, it enforces that all those updates get committed.